docs/site/pages/plugin-implementation.mdx

```diff
@@ -0,0 +1,92 @@
+---
+title: Plugin Implementation
+---
+
+# Plugin Implementation
+
+Plugins main capabilities are to extend or add new functionality by tapping into player's pipeline of components via hooks . In this section we'll go over the steps to implement a plugin.
+
+We'll use the [stage-revert-data](./plugins/stage-revert-data) plugin as example. After creating our plugin class, in the `apply` method we'll add the following code:
+
+```typescript
+export default class StageRevertDataPlugin implements PlayerPlugin {
+  name = 'stage-revert-data-plugin';
+
+  apply(player: Player) {
+    let dataController: DataController;
+    let stageData: String;
+    let commitTransitions: String[];
+    let commitShadowModel: Boolean = false;
+
+    const GatedDataMiddleware = new ValidationMiddleware(
+      () =>
+        commitShadowModel
+          ? undefined
+          : {
+              message: 'staging data',
+              severity: 'error',
+            },
+      { shouldIncludeInvalid: () => true }
+    );
+```
+We defined variables that will be used in the plugin:
+- `dataController`: The data orchestrator of our Player instance
+- `stageData`: State attribute that comes from the view attribute content to enabling the staging of data
+- `commitTransitions`: The list of view names which the shadow model should be committed if transitioned to. Comes from the view state attribute.
+- `commitShadowModel`: Flag that enables committing shadow model into data model after the transition, only if `stageData` is set to true and the next view name is included in `commitTransitions`.
+- `GatedDataMiddleware` Data Middleware that will be used to intercept the data pipeline if `stageData` is set to true.
+
+The next step is to tap into the necessary Player hooks. First we tap into the `viewController` which we can then intercept the `resolveView` hook. By defining our own property method `call` we can intercept the current view state and read the `stageData` and `commitTransitions` properties.
```

Comment:
We might want to clarify that `call` is actually tapable API to intercept calls to the hook [docs](https://github.com/webpack/tapable?tab=readme-ov-file#interception)